### PR TITLE
[kube-prometheus-stack] adding tpl upstream for secretFieldSelector

### PR DESCRIPTION
**Pull request by z-anshun**
#### What this PR does / why we need it
Currently, the "secretFieldSelector" option only supports setting constants. But in more cases we want to be able to set variables (eg `metadata.namespace={{ .Release.Namespace }}`). 
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml [hunk 1]
```diff
@@ -108,7 +108,7 @@ spec:
             - --thanos-ruler-instance-namespaces={{ .Values.prometheusOperator.thanosRulerInstanceNamespaces | join "," }}
             {{- end }}
             {{- if .Values.prometheusOperator.secretFieldSelector }}
-            - --secret-field-selector={{ .Values.prometheusOperator.secretFieldSelector }}
+            - --secret-field-selector={{ tpl (.Values.prometheusOperator.secretFieldSelector) . }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:111]:
```suggestion
            - --secret-field-selector={{ tpl (.Values.prometheusOperator.secretFieldSelector) $ }}
```

**Comment by QuentinBisson** [comment]
This needs a version bump

File: charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml (hunk 1 above)
Comment (z-anshun) [reply charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml]: Can you tell me why '$' is recommended in this way? I found the '.' used in the helm documentation.
https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml]: Sure, `.` is the current scope where `$` is the root scope so if you are in a range or a with, the current scope changes so .Values does not exist in there while `$` allows you to access the root scope

cf. https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml]: Also the other tpl are using the $ in this file so for homogeneity, let's use the same scheme

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**